### ISSUE TITLE
2fa: The SMS fallback option is shown even when a given WP.com account doesn't use SMS as second factor

### ISSUE THREAD

**Issue by xknown** [Automattic/wp-calypso#36809]
We currently show the `Send Code via SMS` button when the 2fa cookie expires or when we want to disable 2fa. 

#### Steps to reproduce
1. Log in to WP.com with an account with 2fa enabled, and with sms disabled (an a8c account for example).
2. Navigate to `/` (`http://calypso.localhost:3000/`)
3. Remove the `twostep_auth` cookie from the `wordpress.com/public-api.wordpress.com` domain.
4. Navigate to `http://calypso.localhost:3000/me`
5. See that the 2fa prompt contains the `Send SMS` button

#### What I expected
The SMS option should likely be only visible when one sets up 2fa with the SMS method.

#### What happened instead
The SMS button is shown, clicking on it doesn't really sends the code because the backend code disables it.

#### Browser / OS version
Chrome, OSX 10.14

#### Screenshot / Video
<img width="448" alt="Screenshot 2019-10-16 at 12 10 05" src="https://user-images.githubusercontent.com/1081870/66909942-f5c3c600-f00d-11e9-8e96-651d00a31312.png">
<img width="675" alt="Screenshot 2019-10-16 at 12 09 55" src="https://user-images.githubusercontent.com/1081870/66909943-f65c5c80-f00d-11e9-9492-cfe5fb22de3c.png">

#### Context / Source
This is somewhat related to #36063, as we probably need to separate the SMS and Authenticator App flows.

**Comment by blackjackkent** [comment]
This is definitely connected to the discussion happening in https://github.com/Automattic/wp-calypso/issues/36063. TL;DR this appears to be a deliberate choice -

```js
// This enables the SMS button on the security key form regardless if we can send SMS or not.
// Otherwise, there's no way to go back to the verification form if smsRequestsAllowed is false.
const shouldEnableSmsButton =
	this.state.smsRequestsAllowed || ( method === 'sms' && twoFactorAuthType === 'webauthn' );
```

- related to the fact that we allow the use of SMS as a fallback if the user doesn't have access to their auth app for whatever reason. #36063 isn't about strictly the same behavior but the discussion is the same (as to what the ultimate flow should be for the user) so we might want to close this ticket as a dupe.

**Comment by xknown** [comment]
@blackjackkent I wrote the comment you cited :), and it's not really related to the fact we use SMS as a fallback. The second part of the above expression was added to fix a UI issue where the "Continue with SMS" button didn't appear because SMS requests are throttled, and the current user chose to use the SMS as second factor.

However, I agree that the discussion in #36063 is related to the fact that we shouldn't be using SMS as fallback.
